docs/contributing/plugins.md

```diff
@@ -0,0 +1,63 @@
+# Adding plugins to trestle
+
+Trestle provides a mechanism for 3rd party providers to extend its command interface via a plugin architecture. All trestle plugins that conforms to this specification will be automatically discovered by trestle if installed, and their command(s) will be added to trestle sub-commands list. Below we describe this plugin mechanism with the help of an example plugin `compliance-trestle-fedramp` that we created as a separate python project that can be installed via `pip`.
+
+## Create the trestle plugin proejct
+
+A separate plugin project needs to be created that will conatin the code for plugin and its commands. This plugin can be given any name and should be available for installation via `pip`. For example, we created a plugin project called  `compliance-trestle-fedramp` which can be installed as `pip install compliance-trestle-fedramp`. The project name doesn't need to start with `compliance-trestle`.
+
+## Project Organization
+
+The plugin project should be organized as shown below.
+
+```text
+compliance-trestle-fedramp
+├── trestle_fedramp
+│   ├── __init.py__
+│   ├── commands
+|   |   ├── __init.py__
+|   |   ├── validate.py
+│   ├── <other source files or folder>
+├── <other files or folder>
+```
+
+Trestle uses naming convention to discover the top-level module of the plugin projects. It expects the top-level module to be named `trestle_{plugin_name}`. In the above example, the top-level module is named as `trestle_fedramp` so that it can be autmatically discovered by trestle. All the python source files should be created inside this module (folder).
```

Comment:
I think this should be "Trestle uses a naming convention". Also I would say for the trestle_{plugin_name} that this convention MUST be followed.

Reply:
done